app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -151,15 +155,24 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
-  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice[]) => {
+  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDeviceFile[]) => {
     const critter = critterData?.find((a) => a.survey_critter_id === survey_critter_id);
-    const critterTelemetryDevice = { ...data[0], critter_id: critter?.critter_id ?? '' };
+    const { attachmentFile, attachmentType, ...critterTelemetryDevice } = {
+      ...data[0],
+      critter_id: critter?.critter_id ?? ''
+    };
     if (telemetryFormMode === TELEMETRY_DEVICE_FORM_MODE.ADD) {
       try {
+        if (attachmentFile && attachmentType === AttachmentType.KEYX) {
+          await bhApi.survey.uploadSurveyKeyx(projectId, surveyId, attachmentFile);
+        } else if (attachmentFile && attachmentType === AttachmentType.OTHER) {
+          await bhApi.survey.uploadSurveyAttachments(projectId, surveyId, attachmentFile);
+        }
         await bhApi.survey.addDeployment(projectId, surveyId, survey_critter_id, critterTelemetryDevice);
         setPopup('Successfully added deployment.');
-      } catch (e) {
-        setPopup('Failed to add deployment.');
+        surveyContext.artifactDataLoader.refresh(projectId, surveyId);
+      } catch (e: any) {
```

Comment:
super nit: lets not use single letter variables. `error` is just better imo.

Also, lets avoid casting stuff to `any` whenever possible. Better to cast it to `Error` (or w/e is appropriate here) as needed

Reply:
For line 191:

I'd prefer `item` over `a`, or `deploymentDataItem` or something. Better yet, is deploymentData of a certain known type? Using that type as part of the variable name can be nice too, just to add some more context.

Dug into it: looks like deploymentData is of type `IAnimalDeployment[]`. So I'd even call it `animalDeployments` or similar, so it is a bit more descriptive of its own type. And then in the map, it could be `animalDeployment` instead of `a`.